src/mechanics/cell/SimpleAssembler.cpp

```diff
@@ -0,0 +1,133 @@
+                                                   CellInterface::MatrixFunction f) const
+{
+    using TripletList = std::vector<Eigen::Triplet<double>>;
+    DofMatrixContainer<TripletList> tripletsJJ;
+    DofMatrixContainer<TripletList> tripletsJK;
+    DofMatrixContainer<TripletList> tripletsKJ;
+    DofMatrixContainer<TripletList> tripletsKK;
+
+    for (NuTo::CellInterface& cell : cells)
+    {
+        const DofMatrix<double> cellHessian = cell.Integrate(f);
+
+        for (DofType dofI : dofTypes)
+        {
+            Eigen::VectorXi numberingDofI = cell.DofNumbering(dofI);
+            for (DofType dofJ : dofTypes)
+            {
+                Eigen::VectorXi numberingDofJ = cell.DofNumbering(dofJ);
+                const Eigen::MatrixXd& cellHessianDof = cellHessian(dofI, dofJ);
+
+                const int numIndependentDofsI = mNumIndependentDofs[dofI];
+                const int numIndependentDofsJ = mNumIndependentDofs[dofJ];
+
+                for (int i = 0; i < numberingDofI.rows(); ++i)
+                {
+                    for (int j = 0; j < numberingDofJ.rows(); ++j)
+                    {
+                        const int globalDofNumberI = numberingDofI[i];
+                        const int globalDofNumberJ = numberingDofJ[j];
+                        const double globalDofValue = cellHessianDof(i, j);
+
+                        const bool activeI = globalDofNumberI < numIndependentDofsI;
+                        const bool activeJ = globalDofNumberJ < numIndependentDofsJ;
+
+                        if (activeI)
+                        {
+                            if (activeJ)
+                            {
+                                tripletsJJ(dofI, dofJ).push_back({globalDofNumberI, globalDofNumberJ, globalDofValue});
+                            }
+                            else
+                            {
+                                tripletsJK(dofI, dofJ)
+                                        .push_back({globalDofNumberI, globalDofNumberJ - numIndependentDofsJ,
+                                                    globalDofValue});
+                            }
+                        }
+                        else
+                        {
+                            if (activeJ)
+                            {
+                                tripletsKJ(dofI, dofJ)
+                                        .push_back({globalDofNumberI - numIndependentDofsI, globalDofNumberJ,
+                                                    globalDofValue});
+                            }
+                            else
+                            {
+                                tripletsKK(dofI, dofJ)
+                                        .push_back({globalDofNumberI - numIndependentDofsI,
+                                                    globalDofNumberJ - numIndependentDofsJ, globalDofValue});
+                            }
+                        } // argh. any better ideas?
```

Comment:
If there were a class like `GlobalDofTripletList`, you could introduce an `operator()` to hide the if-else as you do in `GlobalDofVector::operator()`, but not sure if this is better...